src/main/java/duke/core/Parser.java

```diff
@@ -0,0 +1,241 @@
+package duke.core;
+
+import duke.command.Command;
+import duke.command.AddCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.UpdateCommand;
+
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Represents a <code>Parser</code> that parses user input into a specific 
+ * type of <code>Command</code>. The <code>Parser</code> also parses tasks
+ * related to the command, and checks for illegal instructions.
+ */
+public class Parser {
+    /**
+     * Creates a string from a sub array of a string array.
+     *
+     * @param words The string array to be processed.
+     * @param start The starting position in the string array.
+     * @param end The ending position in the string array.
+     * @return A string consisting of all the words from the starting position
+     *          to the ending position, separated by spaces.
+     */
+    private static String subString(String[] words, int start, int end) {
+        StringBuilder sb = new StringBuilder();
+        IntStream.rangeClosed(start, end - 1).forEach(i -> sb.append(words[i] + " "));
+        return sb.toString().trim();
+    }
+
+    /**
+     * Finds the index of a given string in a string array.
+     *
+     * @param words The string array to be inspected.
+     * @param s The string to be searched.
+     * @return The index of the given string in the string array. Returns -1 if
+     *          the given string does not occur in the array.
+     */
+    private static int findIdx(String[] words, String s) {
+        return Stream.of(words).collect(Collectors.toList()).indexOf(s);
+    }
+
+    private static String intToOrdinal(int i) {
+        assert i >= 1 && i <= 31 : "Days of the month should be in the range 1 to 31";
+        if (i == 11 || i == 12 || i == 13) {
+            return i + "th";
+        } else {
+            switch (i % 10) {
+            case 1:
+                return i + "st";
+            case 2:
+                return i + "nd";
+            case 3:
+                return i + "rd";
+            default:
+                return i + "th";
+            }
+        }
+    }
+
+    private static String timeToAmPm(LocalDateTime dateTime) {
```

Comment:
Can consider using the SimpleDateFormat class instead of re-inventing the wheel.